source/extensions/filters/listener/http_inspector/http_inspector.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "envoy/event/file_event.h"
+#include "envoy/event/timer.h"
+#include "envoy/network/filter.h"
+#include "envoy/stats/scope.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "common/common/logger.h"
+
+#include "absl/container/flat_hash_set.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace HttpInspector {
+
+/**
+ * All stats for the http inspector. @see stats_macros.h
+ */
+#define ALL_HTTP_INSPECTOR_STATS(COUNTER)                                                          \
+  COUNTER(read_error)                                                                              \
+  COUNTER(http10_found)                                                                            \
+  COUNTER(http11_found)                                                                            \
+  COUNTER(http2_found)                                                                             \
+  COUNTER(http_not_found)
+
+/**
+ * Definition of all stats for the Http inspector. @see stats_macros.h
+ */
+struct HttpInspectorStats {
+  ALL_HTTP_INSPECTOR_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+/**
+ * Global configuration for http inspector.
+ */
+class Config {
+public:
+  Config(Stats::Scope& scope);
+
+  const HttpInspectorStats& stats() const { return stats_; }
+
+  static constexpr uint32_t MAX_INSPECT_SIZE = 1024;
```

Comment:
Reading spec a bit more, https://tools.ietf.org/html/rfc7230#section-3.1.1 says:

>    Various ad hoc limitations on request-line length are found in practice.  It is RECOMMENDED that all HTTP senders and recipients support, at a minimum, request-line lengths of 8000 octets.

So let's set this somewhere above that, 8192? @PiotrSikora WDYT?

Reply:
Sounds reasonable.